include/envoy/http/filter.h

```diff
@@ -224,6 +224,12 @@ class StreamFilterBase {
   virtual ~StreamFilterBase() {}
 
   /**
+   * This routine is called on filter creation, allowing filters to subscribe to watermark events
+   * on the downstream stream and downstream connection.
+   */
+  virtual void provideWatermarkCallbacks(Http::DownstreamWatermarkProvider& provider) PURE;
```

Comment:
I guess I don't understand the value of the connection manager calling this method? Why not just add `void add DownstreamWatermarkCallbacks Callbacks(DownstreamWatermarkCallbacks& callbacks)` on the filter callbacks and avoid every filter having to implement this method?

If this stays: nit: Can lose `Http:` prefix since inside HTTP namespace.